persistence/context.go

```diff
@@ -36,18 +36,22 @@ type PostgresContext struct {
 	networkId string
 }
 
-func (p *PostgresContext) NewSavePoint(bytes []byte) error {
-	p.logger.Info().Bool("TODO", true).Msg("NewSavePoint not implemented")
+// NewSavePoint generates a new Savepoint for this context.
+func (p *PostgresContext) NewSavePoint() error {
+	if err := p.stateTrees.Savepoint(); err != nil {
+		return err
+	}
 	return nil
 }
 
-// TECHDEBT(#327): Guarantee atomicity betweens `prepareBlock`, `insertBlock` and `storeBlock` for save points & rollbacks.
-func (p *PostgresContext) RollbackToSavePoint(bytes []byte) error {
-	p.logger.Info().Bool("TODO", true).Msg("RollbackToSavePoint not fully implemented")
-	return p.tx.Rollback(context.TODO())
+// RollbackToSavepoint triggers a rollback for the current pgx transaction and the underylying submodule stores.
+func (p *PostgresContext) RollbackToSavePoint() error {
+	ctx := context.TODO()
```

Comment:
```suggestion
	ctx, _ := p.getCtxAndTx
```
Unsure if this will affect things but each `PostgresContext` already has a `context.Context` so we could reuse it?

Reply:
I initially avoided the getCtxAndTx method because of the fact we haven't addressed context handling yet, but it shouldn't effect anything to use the getCtxAndTx method here. I will add a TECHDEBT comment about the context handling, too.